### PR TITLE
Add setters and getters for the target configuration ID associated with a TaskInfo

### PR DESCRIPTION
**Pull request by gabrielhartmann**
Configurations are stored in the ConfigStore with UUID's for keys.  These UUIDs are set as labels on TaskInfo objects to indicate what configuration a TaskInfo is associated with.

### CODE REVIEW

File: src/main/java/org/apache/mesos/offer/TaskUtils.java [hunk 1]
```diff
@@ -94,6 +93,61 @@ public static TaskInfo clearTransient(TaskInfo taskInfo) {
...
+    for (Label label : taskInfo.getLabels().getLabelsList()) {
+      if (label.getKey().equals(TARGET_CONFIGURATION_KEY)) {
+        return UUID.fromString(label.getValue());
+      }
+    }
+
+    throw new TaskException("TaskInfo does not contain label with key: " + TARGET_CONFIGURATION_KEY);
+  }
+
+  private static TaskInfo clearConfigTarget(TaskInfo taskInfo) {
```
Comment (mrbrowning) [inline src/main/java/org/apache/mesos/offer/TaskUtils.java]: Nit: the term is reversed here ("ConfigTarget" vs "TargetConfiguration" elsewhere) and could maybe reference more clearly the fact that it's removing labels? Maybe `clearTargetConfigLabel`.

File: src/test/java/org/apache/mesos/offer/TaskUtilsTest.java [hunk 2]
```diff
@@ -21,7 +26,37 @@ public void testInvalidToTaskName() throws Exception {
         TaskUtils.toTaskName(getTaskId(testTaskName + "_id"));
     }
 
+    @Test(expected=TaskException.class)
+    public void testGetTargetConfigurationFailure() throws Exception{
```
Comment (mrbrowning) [inline src/test/java/org/apache/mesos/offer/TaskUtilsTest.java]: Space before brace?

File: src/test/java/org/apache/mesos/offer/TaskUtilsTest.java [hunk 3]
```diff
@@ -21,7 +26,37 @@ public void testInvalidToTaskName() throws Exception {
...
+        TaskUtils.getTargetConfiguration(getTestTaskInfo());
+    }
+
+    @Test
+    public void testGetTargetConfigurationSuccess() throws Exception{
```
Comment (mrbrowning) [inline src/test/java/org/apache/mesos/offer/TaskUtilsTest.java]: See above.

File: src/main/java/org/apache/mesos/offer/TaskUtils.java [hunk 4]
```diff
@@ -13,11 +10,13 @@
 import org.apache.mesos.Protos.TaskID;
 import org.apache.mesos.Protos.TaskInfo;
 import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.config.ConfigStore;
 
 /**
  * Various utility methods for manipulating data in {@link TaskInfo}s.
  */
 public class TaskUtils {
+  public static final String TARGET_CONFIGURATION_KEY = "target_configuration";
```
Comment (nickbp) [inline src/main/java/org/apache/mesos/offer/TaskUtils.java]: Can this be private?

Comment (gabrielhartmann) [inline src/main/java/org/apache/mesos/offer/TaskUtils.java]: Yes.  Done.

**Comment by gabrielhartmann** [comment]
@nickbp and @mrbrowning: Comments addressed.

**Pull request merged**